examples/paywall-tester/src/main/java/com/revenuecat/paywallstester/MainActivity.kt

```diff
@@ -1,17 +1,25 @@
 package com.revenuecat.paywallstester
 
 import android.os.Bundle
+import android.util.Log
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
 import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.material3.MaterialTheme
 import androidx.compose.material3.Surface
 import androidx.compose.ui.Modifier
 import com.revenuecat.paywallstester.ui.theme.PaywallTesterAndroidTheme
+import com.revenuecat.purchases.Offering
+import com.revenuecat.purchases.ui.revenuecatui.activity.PaywallActivityLauncher
+import com.revenuecat.purchases.ui.revenuecatui.activity.PaywallResult
+import com.revenuecat.purchases.ui.revenuecatui.activity.PaywallResultHandler
+
+class MainActivity : ComponentActivity(), PaywallResultHandler {
+    private lateinit var paywallActivityLauncher: PaywallActivityLauncher
 
-class MainActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        paywallActivityLauncher = PaywallActivityLauncher(this, this)
```

Comment:
What if we make PaywallResultHandler a lambda like this:

```
typealias PaywallResultHandler = (PaywallResult) -> Unit
```

Then in the constructor of the launcher, we can handle the `ActivityResultCallback` there directly:

```
    constructor(activity: ComponentActivity, resultHandler: PaywallResultHandler) {
        activityResultLauncher = activity.registerForActivityResult(PaywallContract()) { paywallResult ->
            resultHandler(it)
        }
    }
```

Then, the activity can be started like this:

```
paywallActivityLauncher = PaywallActivityLauncher(this) { result ->
    Log.e("PaywallsTester", "LAUNCH PAYWALL RESULT: $result")
}
```

This way, folks don't have to make their Activity conform to `PaywallResultHandler` (`ActivityResultCallback`) and override `onActivityResult`.

Reply:
I thought about that... My reasoning to go this way is in case we ever want to add another callback or anything, this would be more extensible even though slightly harder to use... Maybe it's ok though. We can discuss that

Reply:
For adding another callback, we could have it as:

```
    constructor(activity: ComponentActivity, resultHandler: PaywallResultHandler, anotherCallback: NewCallback) {
```

And then we could do ourselves the complicated part, which is figuring out which callback to call using the activity result code. Otherwise devs would have to handle that

Reply:
Well, that's not as extensible IMO... Another point to consider is that this `PaywallActivityLauncher` needs to be created during the `onCreate`, it can't be created after (it will fail at runtime if you try). In that case, you would have a callback at the activity level, not inside a method, so you still need to reference some other class there. I think for a class variable, an interface is a more common pattern... What do you think?

Reply:
Makes sense. Using the interface is actually more flexible + it's using a "system" interface, that devs will probably be familiar with. Let's do the activity